### ISSUE TITLE
sample univoice : Peer cannot send and receive

### ISSUE THREAD

**Issue by airqleon1** [adrenak/univoice-sample#2]
Peer 0 and Peer 1 are working fine, Peer 0 and Peer 2 are working fine, but Peer 1 and Peer 2 are not working fine

**Comment by adrenak** [comment]
Hey, are you testing with PUN network?

**Comment by airqleon1** [comment]
UniVoiceTelepathyNetwork.New(8987)

**Comment by adrenak** [comment]
Having a look, it's a new network implementation so I'm not surprised

**Comment by airqleon1** [comment]
I looked at unvoice-telepathy-network and it seems like the server is not broadcasting all peers

**Comment by adrenak** [comment]
yes likely that is the case, the if peer 1 has to send data to peer 2 (they are both clients), it is first sent to peer 0 (the server) which then forwards it to peer 2.

I'll have a look at stuff, the last few commits across univoice have been done in a hurry to get some features implemented for a sponsor. That's why the docs haven't been updated yet.

**Comment by airqleon1** [comment]
The problem of server.Send(recipient, new ArraySegment<byte>(segmentBytes)); in unvoice-telepathy-network, segmentBytes is not a packet.
Thank you

**Comment by shahiltp** [comment]
whats the change in that line
do we need to create byteswriter packet there

**Comment by chwar** [comment]
Hey, I'm having the same issue with telepathy. The peers cannot communicate between each other. 
I require my voice chat to be self hosted, so I can't use Photon. Any news on this? Thanks :)

**Comment by adrenak** [comment]
Hey everyone, this issue went ignored for a while. I believe some related fixes were posted somewhere, I'll have a look tomorrow and fix. Thanks for being patient!

**Comment by chwar** [comment]
That sounds great. Thanks for replying!
